Library/PTPusher.m

```diff
@@ -274,6 +274,20 @@ - (void)subscribeAll
   }
 }
 
+- (void)unsubscribeAllChannels
+{
+  if (channels && channels.count) {
+	NSMutableArray *channelsToUnsubscribe = [[NSMutableArray alloc] init];
+	  for (PTPusherChannel *channel in [channels allValues]) {
+		[channelsToUnsubscribe addObject:channel];
```

Comment:
When you call `[channel unsubscribe]`, that channel will be removed from the `channels` array, which will cause a crash in the loop (the array size cannot change in the middle of the loop)

This crash will only happen when there are at least 2 channels to be unsubscribed.

Please refer to [this Stackoverflow thread](http://stackoverflow.com/questions/111866/best-way-to-remove-from-nsmutablearray-while-iterating) to for more info.